prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -158,19 +179,17 @@ def build_transmission():
     hifld_data_dir = os.path.join(os.path.dirname(__file__), "..", "data")
     hifld_zones = get_zone(os.path.join(hifld_data_dir, "zone.csv"))  # noqa: F841
 
-    # Filter substations
-    substations_with_lines = filter_substations_with_zero_lines(hifld_substations)
-    check_for_location_conflicts(substations_with_lines)
+    # Filter substations based on their `LINES` attribute, check for location dupes
+    substations = filter_substations_with_zero_lines(hifld_substations)
+    check_for_location_conflicts(substations)
 
-    # Filter lines
-    lines_with_substations = filter_lines_with_unavailable_substations(hifld_lines)
-    lines_with_matching_substations = filter_lines_with_no_matching_substations(
-        lines_with_substations, substations_with_lines
-    )
-    lines_with_matching_substations = filter_lines_with_nonmatching_substation_coords(
-        lines_with_matching_substations, substations_with_lines
-    )
-    lines_with_matching_substations = filter_lines_with_identical_substation_names(
-        lines_with_matching_substations
-    )
-    return lines_with_matching_substations
+    # Filter lines based on substations
+    lines = filter_lines_with_unavailable_substations(hifld_lines)
+    lines = filter_lines_with_no_matching_substations(lines, substations)
+    lines = filter_lines_with_nonmatching_substation_coords(lines, substations)
+    lines = filter_lines_with_identical_substation_names(lines)
+
+    # Filter substations based on lines
+    substations = filter_substations_by_connected_components(lines, substations)
+
+    return lines, substations
```

Comment:
So now, it returns transmission lines with all components + substations within the largest component, right? What do we do with it? Those substations within other components are permanently dropped?

Reply:
The other ones seem to be permanently dropped:
https://github.com/Mingze-Li128/PreREISE/blob/410dabaf7a0afb85b4ade854251a3ac789a8d662/prereise/gather/hiflddata/data_trans.py#L954

Reply:
That's my read as well. Then the question is should we proceed in this way or should we start a thread with our collaborators to confirm we won't get much value from the dropped substations? Could you remind me the following strategy with lines from all components + substations from the biggest component?

Reply:
That's a good question. Looking at the original `data_transform`, it does seem like we filter the substations but not the lines, but I don't recall the logic. Maybe @terrywqf has more context.